src/testing/mock.md

```diff
@@ -1,218 +1,310 @@
 
 ```rust
-use primitives::{BuildStorage, traits::IdentityLookup, testing::{Digest, DigestItem, Header, UintAuthorityId}};
-use srml_support::impl_outer_origin;
-use runtime_io;
-use substrate_primitives::{H256, Blake2Hasher};
+use support::{impl_outer_event, impl_outer_origin};
+
+impl_outer_origin!{
+	pub enum Origin for Runtime {}
+}
 ```
 
-It is also necessary to import the module configuration traits. For this recipe,  import `Module` and `Trait`. We may also import `GenesisConfig` if some storage items are set to be configured in the genesis block (marked with `config()` in the `decl_storage` block).
+It is possible to placed signed transactions as parameters in runtime methods that require the `origin` input. See the [full code in the kitchen](https://github.com/substrate-developer-hub/recipes/tree/master/kitchen/modules/hello-substrate)), but this looks like
 
 ```rust
-use crate::{GenesisConfig, Module, Trait};
+#[test]
+fn last_value_updates() {
+	ExtBuilder::build().execute_with(|| {
+		HelloSubstrate::set_value(Origin::signed(1), 10u64);
+		// some assert statements
+	})
+}
 ```
 
-### Constructing a Mock Runtime
+Run `cargo test` to run the unit tests
+
+## Storage Changes
 
-To test the module, construct a mock runtime. To do so, create a configuration type called `Test` which implements the configuration traits.
+Changes to storage can be checked by direct calls to the storage values. The syntax is the same as it would be in the module's runtime methods
 
 ```rust
-#[derive(Clone, PartialEq, Eq, Debug)]
-pub struct Test;
+#[test]
+fn last_value_updates() {
+	ExtBuilder::build().execute_with(|| {
+	HelloSubstrate::set_value(Origin::signed(1), 10u64);
+	assert_eq!(HelloSubstrate::last_value(), 10u64);
+	HelloSubstrate::set_value(Origin::signed(2), 11u64);
+	assert_eq!(HelloSubstrate::last_value(), 11u64);
+	// .. more assert statements
+	})
+}
 ```
 
-The [derive attribute](https://doc.rust-lang.org/edition-guide/rust-2018/macros/custom-derive.html) ensures that you don't have to manually implement the `Clone, PartialEq, Eq, Debug` traits; the compiler does this for you thereby ensuring that the `Test` type conforms to the behavior of these traits.
-
-Even so, this doesn't work for all traits. Indeed, there are a few traits that require manual implementation to set up the testing environment in `test.rs`. In most case, these *implementations* are limited to specifying the type in your module that corresponds to the type in the imported module. For example, the [Staking module](https://github.com/paritytech/substrate/blob/master/srml/staking/src/mock.rs) implements the `balances` trait in its `mock.rs` file like so:
+For context, the tested module's `decl_storage` block looks like
```

Comment:
It seems like all the context here is distracting. What about a simpler generic test like

```rust
#[test]
fn last_value_updates() {
	ExtBuilder::build().execute_with(|| {
	let expected_value = ...;
	assert_eq!(HelloSubstrate::last_value(), expected_value);
	
```

Not sure which is better. Up to you.

Reply:
I made it like that and added the necessary `set_value` call.